client-server-client/src/main/scala/zio/raft/client/RaftClient.scala

```diff
@@ -0,0 +1,572 @@
+      .merge(keepAliveStream)
+      .merge(timeoutStream)
+    
+    // Pure functional state machine: state handles events
+    unifiedStream
+      .runFold(initialState) { (state, event) =>
+        // State handles everything - just pass dependencies
+        state.handle(event, transport, config, serverRequestQueue)
+      }
+      .unit
+  }
+}
+
+/**
+ * Functional state machine: each state handles events and returns new states.
+ */
+sealed trait ClientState {
+  def stateName: String
+  def handle(
+    event: StreamEvent,
+    transport: ZmqClientTransport,
+    config: ClientConfig,
+    serverRequestQueue: Queue[ServerRequest]
+  ): UIO[ClientState]
+}
+
+object ClientState {
+  
+  case object Disconnected extends ClientState {
+    override def stateName: String = "Disconnected"
+    
+    override def handle(
+      event: StreamEvent,
+      transport: ZmqClientTransport,
+      config: ClientConfig,
+      serverRequestQueue: Queue[ServerRequest]
+    ): UIO[ClientState] = {
+      event match {
+        case StreamEvent.Action(ClientAction.Connect) =>
+          // Reconnect: generate nonce and transition to ConnectingNewSession
+          for {
+            nonce <- Nonce.generate()
+            now <- Clock.instant
+            address = config.clusterAddresses.head
+            _ <- transport.connect(address).orDie
+            _ <- transport.sendMessage(CreateSession(config.capabilities, nonce)).orDie
+          } yield ConnectingNewSession(
+            capabilities = config.capabilities,
+            nonce = nonce,
+            addresses = config.clusterAddresses,
+            currentAddressIndex = 0,
+            createdAt = now
+          )
+        
+        case StreamEvent.ServerMsg(message) =>
+          ZIO.logWarning(s"Received message while disconnected: ${message.getClass.getSimpleName}").as(this)
+        
+        case StreamEvent.Action(ClientAction.SubmitCommand(_, promise)) =>
+          promise.fail(new IllegalStateException("Not connected")).ignore.as(this)
+        
+        case _ =>
+          ZIO.succeed(this)
+      }
+    }
+  }
+  
+  /**
+   * Connecting to create a NEW session (no existing sessionId).
+   */
+  case class ConnectingNewSession(
+    capabilities: Map[String, String],
+    nonce: Nonce,
+    addresses: List[String],
+    currentAddressIndex: Int,
+    createdAt: Instant
+  ) extends ClientState {
+    override def stateName: String = "ConnectingNewSession"
+    
+    override def handle(
+      event: StreamEvent,
+      transport: ZmqClientTransport,
+      config: ClientConfig,
+      serverRequestQueue: Queue[ServerRequest]
+    ): UIO[ClientState] = {
+      event match {
+        case StreamEvent.ServerMsg(SessionCreated(sessionId, responseNonce)) =>
+          if (nonce == responseNonce) {
+            for {
+              _ <- ZIO.logInfo(s"Session created: $sessionId")
+              now <- Clock.instant
+              nextRequestId <- Ref.make(RequestId.zero)
```

Comment:
nextReqiestId ref should be a parameter to the ConnectingState, as we do accept requests when we are in connecting state.

Here we should also send all pending requests.